src/parser/Statement.ts

```diff
@@ -3536,3 +3576,188 @@ export class AliasStatement extends Statement {
+    public readonly range: Range | undefined;
+
+    transpile(state: BrsTranspileState) {
+        let results = [] as TranspileResult;
+        //if   (already indented by block)
+        if (!state.conditionalCompileStatement) {
+            // only transpile the #if in the case when we're not in a conditionalCompileStatement already
+            results.push(state.transpileToken(this.tokens.hashIf ?? createToken(TokenKind.HashIf)));
+        }
+
+        results.push(' ');
+        //conditions
+        if (this.tokens.not) {
+            results.push('not');
+            results.push(' ');
+        }
+        results.push(state.transpileToken(this.tokens.condition));
+        state.lineage.unshift(this);
+
+        //if statement body
+        let thenNodes = this.thenBranch.transpile(state);
+        state.lineage.shift();
+        if (thenNodes.length > 0) {
+            results.push(thenNodes);
+        }
+        //else branch
+        if (this.elseBranch) {
+            const elseIsCC = isConditionalCompileStatement(this.elseBranch);
+            const endBlockToken = elseIsCC ? (this.elseBranch as ConditionalCompileStatement).tokens.hashIf ?? createToken(TokenKind.HashElseIf) : this.tokens.hashElse;
+            //else
+
+            results.push(...state.transpileEndBlockToken(this.thenBranch, endBlockToken, createToken(TokenKind.HashElse).text));
+
+            if (elseIsCC) {
+                //chained else if
+                state.lineage.unshift(this.elseBranch);
+
+                // transpile following #if with knowledge of current
+                const existingCCStmt = state.conditionalCompileStatement;
+                state.conditionalCompileStatement = this;
+                let body = this.elseBranch.transpile(state);
+                state.conditionalCompileStatement = existingCCStmt;
+
+                state.lineage.shift();
+
+                if (body.length > 0) {
+                    //zero or more spaces between the `else` and the `if`
+                    results.push(...body);
+
+                    // stop here because chained if will transpile the rest
+                    return results;
+                } else {
+                    results.push('\n');
+                }
+
+            } else {
+                //else body
+                state.lineage.unshift(this.tokens.hashElse!);
+                let body = this.elseBranch.transpile(state);
+                state.lineage.shift();
+
+                if (body.length > 0) {
+                    results.push(...body);
+                }
+            }
+        }
+
+        //end if
+        results.push(...state.transpileEndBlockToken(this.elseBranch ?? this.thenBranch, this.tokens.hashEndIf, '#end if'));
+
+        return results;
+    }
+
+    walk(visitor: WalkVisitor, options: WalkOptions) {
+        if (options.walkMode & InternalWalkMode.walkStatements) {
+            let conditionTrue = options.bsConsts?.get(this.tokens.condition.text.toLowerCase());
```

Comment:
Shouldn't we fall back to the parent bsConsts if not provided?
```typescript
const bsConsts = options.bsConsts ?? this.getBsConsts();
```
That way it supports calling .walk on any ast node.